contracts/token/ERC20/extensions/draft-ERC20FlashMint.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./draft-IERC3156.sol";
+import "../ERC20.sol";
+
+/**
+ * @dev Implementation of the ERC3156 Flash loans extension, as defined in
+ * https://eips.ethereum.org/EIPS/eip-3156[ERC-3156].
+ *
+ * Adds the {flashLoan} method, which provide flash loan support, at the token
+ * level, with no fee.
+ */
+abstract contract ERC20FlashMint is ERC20, IERC3156FlashLender {
+    bytes32 constant internal RETURN_VALUE = keccak256("ERC3156FlashBorrower.onFlashLoan");
+
+    /**
+     * @dev Return the maximum amount of token available for loan.
+     * @param token The address of the token that is requested.
+     * @return The amont of token that can be loaned.
+     */
+    function maxFlashLoan(address token) public view override returns (uint256) {
+        return token == address(this) ? type(uint256).max - ERC20.totalSupply() : 0;
```

Comment:
I was under the impression that, since `totalSupply` can on overridden, and what we really care about is the `mint` not overflowing if someone wanted to flashMint, then te best way to protect `_mint` overflowing is to use the internal bookkeeping variables.

But If you'd rather use the overloaded version I have no strong argument. (worst case scenario some flashMint call revert, but that would not break the ERC20 token.